cli/azd/pkg/ext/hooks_manager_test.go

```diff
@@ -103,11 +105,17 @@ func ensureScriptsExist(t *testing.T, configs map[string]*HookConfig) {
 	for _, hook := range configs {
 		ext := filepath.Ext(hook.Run)
 
-		if ext != "" {
+		if isValidFileExtension(ext) {
 			err := os.MkdirAll(filepath.Dir(hook.Run), osutil.PermissionDirectory)
 			require.NoError(t, err)
 			err = os.WriteFile(hook.Run, nil, osutil.PermissionExecutableFile)
 			require.NoError(t, err)
 		}
 	}
 }
+
+var fileExtensionRegex = regexp.MustCompile(`^\.[\w]{1,4}$`)
```

Comment:
I am curious here about why are we blocking `>4` extensions??   
Might be good to have a comment about why that's invalid for azd.
Would it make sense to expect either .sh/.bash or .ps1 ?   
Should .mp3 be valid according to the regex, but `.torrent` be invalid?

Reply:
FWIW - this is in test code.  I think this just makes it easy for Wallace to write tests that check both the "run an existing file" and the "run this inline script".

The semantics for the product seem to be "if the value of run is a relative path to a script file that exists (regardless of slashes, then execute that script, otherwise treat the value of run as an inline script".